src/storage/conversion/ConversionUtil.ts

```diff
@@ -12,15 +12,30 @@ import type { RepresentationConverterArgs } from './RepresentationConverter';
  * @throws UnsupportedHttpError
  * If the type preferences are undefined.
  *
- * @returns The filtered list of preferences.
+ * @returns The weighted and filtered list of supported types.
  */
 export const matchingTypes = (preferences: RepresentationPreferences, supported: string[]):
 RepresentationPreference[] => {
   if (!Array.isArray(preferences.type)) {
     throw new UnsupportedHttpError('Output type required for conversion.');
   }
-  return preferences.type.filter(({ value, weight }): boolean => weight > 0 &&
-    supported.some((type): boolean => matchingMediaType(value, type)));
+
+  // Map all supported types to their corresponding weight
+  const weightedSupported = supported.map((type): RepresentationPreference => {
+    let weight = 0;
+    for (const preference of preferences.type!) {
+      if (matchingMediaType(type, preference.value)) {
+        // Never accept anything with weight 0
+        if (preference.weight === 0) {
```

Comment:
Given that it is a floating-point value, perhaps…

```suggestion
        if (preference.weight < 0.001) {
```

Reply:
Floating points should have no issue representing 0 though?

Reply:
Some more thoughts about this: although accept headers can only go as low as 0.001, if we ever start combining preferences (converters having preferences on what they can do better/easier) it would be possible to reach lower values and I don't think we want to equate a lower value to "not acceptable".

Reply:
>  it would be possible to reach lower values and I don't think we want to equate a lower value to "not acceptable".

Ah I just thought the opposite, given that `qvalue` is defined as a three-digit number. But I'm fine with a hard zero.